project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/productListMacro.html.twig

```diff
@@ -1,61 +1,60 @@
-{% macro list(productDetails, listCssClass, productHeadingTagName) %}
+{% macro list(products, listCssClass, productHeadingTagName) %}
     {% import '@ShopsysShop/Front/Inline/Product/productFlagsMacro.html.twig' as productFlags %}
     {% set domain = getDomain() %}
 
     <ul class="list-products js-product-list {{ listCssClass|default('') }}">
-        {% for productDetail in productDetails %}
-            {% set product = productDetail.product %}
-
+        {% for product in products %}
             <li class="list-products__item js-list-products-item">
                 <div class="list-products__item__in">
-                    <a href="{{ url('front_product_detail', {'id': productDetail.product.id}) }}" class="list-products__item__block">
+                    <a href="{{ url('front_product_detail', {'id': product.id}) }}" class="list-products__item__block">
                         <{{ productHeadingTagName|default('h2') }} class="list-products__item__title js-list-products-item-title">
-                            {{ productDetail.product.name }}
+                            {{ product.name }}
                         </{{ productHeadingTagName|default('h2') }}>
 
                         <div class="list-products__item__image">
-                            {{ image(productDetail.product, { size: 'list', alt: productDetail.product.name }) }}
-                            {{ productFlags.list(productDetail.product, 'in-flag--in-list') }}
+                            {{ image(product, { size: 'list', alt: product.name }) }}
+                            {{ productFlags.list(product, 'in-flag--in-list') }}
                         </div>
 
                         <div class="list-products__item__info">
                             <p class="list-products__item__info__description">
                                 {{ product.getShortDescription(domain.id) }}
                             </p>
-                            {% if productDetail.sellingPrice is not null %}
+                            {% set productSellingPrice = getProductSellingPrice(product) %}
```

Comment:
There are exceptions thrown when you access the property the bad way :slightly_smiling_face: At the moment I thought it was a good idea but it could be solved some other way (eg. pass the data into template from controller as a separate variable, let us discuss it personally).